nix/modules/lanzaboote.nix

```diff
@@ -15,12 +11,43 @@ let
   loaderConfigFile = loaderSettingsFormat.generate "loader.conf" cfg.settings;
 
   configurationLimit = if cfg.configurationLimit == null then 0 else cfg.configurationLimit;
+
+  loaderKeyOpts = { ... }:
+    let
+      mkAuthOption = variableName: mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = "Auth variable file for ${variableName}";
+      };
+    in
+    {
+      options = {
+        db = mkAuthOption "db";
+        KEK = mkAuthOption "KEK";
+        PK = mkAuthOption "PK";
+      };
+    };
 in
 {
   options.boot.lanzaboote = {
     enable = mkEnableOption "Enable the LANZABOOTE";
 
-    enrollKeys = mkEnableOption "Automatic enrollment of the keys using sbctl";
+    safeAutoEnroll = mkOption {
+      type = types.nullOr (types.submodule loaderKeyOpts);
+      default = null;
+      description = ''
+        Perform safe automatic (or manual) enrollment of Secure Boot variables
+        via .auth variables.
+
+        Files will be put in /loader/keys/auto/{db,KEK,PK}.auth.
+
+        If you are using systemd-boot, they will be enrolled if it's deemed safe
+        or `secure-boot-enroll` is set to `force`. Usually, detected virtual machine environments
```

Comment:
How do you set `secure-boot-enroll` this to force? Is this an efi variable?

Reply:
Documented, it's in the configuration file.